terraform-aws-github-runner/modules/runners/lambdas/runners/src/lambda.ts

```diff
@@ -1,24 +1,64 @@
-import { Context, SQSEvent, ScheduledEvent } from 'aws-lambda';
+import { SQS } from 'aws-sdk';
+import { Context, SQSEvent, SQSRecord, ScheduledEvent } from 'aws-lambda';
 
+import { Config } from './scale-runners/config';
 import { scaleDown as scaleDownR } from './scale-runners/scale-down';
-import { scaleUp as scaleUpR, RetryableScalingError } from './scale-runners/scale-up';
+import { scaleUp as scaleUpR, RetryableScalingError, ActionRequestMessage } from './scale-runners/scale-up';
+import { getDelayWithJitter } from './scale-runners/utils';
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export async function scaleUp(event: SQSEvent, context: Context, callback: any) {
   console.dir(event, { depth: 5 });
   try {
-    for (const e of event.Records) {
-      await scaleUpR(e.eventSource, JSON.parse(e.body));
+    const evtFailed: Array<SQSRecord> = [];
+
+    for (const evt of event.Records) {
+      try {
+        await scaleUpR(evt.eventSource, JSON.parse(evt.body));
+      } catch (e) {
+        if (e instanceof RetryableScalingError) {
+          console.error(`Retryable error thrown: "${e.message}"`);
+          evtFailed.push(evt);
+        } else {
+          throw e;
```

Comment:
A non-retryable error will fail the whole batch (`evtFailed` won't be processed). Is that the desired behavior?

Reply:
yes, then the best that can be done is retry the whole batch, because the failure is not easily recoverable. The message should return to the queue and be retried. In this case it follows the SQS retry-delivery policy

Reply:
I see. Thanks for the clarification.